.github/workflows/code-quality.yaml

```diff
@@ -0,0 +1,96 @@
+name: ✨ Code Quality Check
+on:
+  push:
+    branches:
+      - main
+      - dev
+  pull_request: {}
```

Comment:
If we use on.push, the CI check will run after merging into the main branch, which doesn't seem quite right. We want the CI workflow to run before merging. 

If it succeeds, then merging should be allowed; if it fails, merging should be blocked.

Reply:
I would say it's desired to run check after merging, you never know what happen and whether the code is functioning after a complex merge.
This would also run on every pull request push, which includes the last one before merging.
blocking is set by project owner in setting, I don't have access to that.

Reply:
I agree that a check after the merge is necessary,

I think there should be two checks,

1. Pre-merge check (whether the current code changes are legal)
2. Post-merge check (whether the merged code is legal)

But the ci yaml here mainly seems to be for the first check, 
if it's for the secondar check, should it be placed in the deploy yaml's job, such as

```yaml
on
  push
     - branches
        - main
        - dev
        - 
jobs
    test:
         ...steps

    deploy:
         needs:[test]
         ...steps
    ...other jobs

```